### PR TITLE
test(multi-action-button): add accessibility tests

### PR DESCRIPTION
**Pull request by ZhuoyuJin**
### Proposed behaviour
- Add `accessibility` Cypress tests for the `Multi-action-button` component to use the new cypress-component-react framework for testing.

### Current behaviour
Old approach for accessibility tests

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- [x] Run `npx cypress open --component` to check if there are newly added tests
- [x] Check if the `multi-action-button.test.js` file passed
- [x] Run `npx cypress run --component` to check none of the other *.test.js files have regressed
- [x] Run `npx cypress run --e2e` to check none of the feature files have regressed
- [x] Run `npm run build-storybook` -> `npx sb extract` -> `npx http-server storybook-static -p 9001` and run `npx cypress run --e2e` and check that `Multi-action-button` tests are not running twice.

### CODE REVIEW

File: src/components/multi-action-button/multi-action-button-test.stories.tsx [hunk 1]
```diff
@@ -76,3 +77,28 @@ MultiActionButtonStory.story = {
     text: "Example Multi Action Button",
   },
 };
+
```
Comment (DlgSHi) [inline src/components/multi-action-button/multi-action-button-test.stories.tsx:80]: can you add `includeStories` parameter to include only `default` story, please?

**Review by DlgSHi — CHANGES_REQUESTED** [review]

File: src/components/multi-action-button/multi-action-button.test.js [hunk 2]
```diff
@@ -538,4 +516,60 @@ describe("should check colors for MultiActionButton component", () => {
...
+      CypressMountWithProviders(<stories.Subtext />);
+
+      cy.checkAccessibility();
+    });
+
+    it("should pass accessibility tests for MultiActionButton InOverflowHiddenContainer story", () => {
+      CypressMountWithProviders(<stories.InOverflowHiddenContainer />);
+
+      cy.checkAccessibility();
+    });
```
Comment (stephenogorman) [inline src/components/multi-action-button/multi-action-button.test.js:573]: You could add an accessibility test for the MultiActionNestedInDialog const.

**Review by DlgSHi — APPROVED** [review]

**Review by stephenogorman — DISMISSED** [review]

**Review by stephenogorman — APPROVED** [review]

**Review by DlgSHi — CHANGES_REQUESTED** [review]
1. We need to expand the hidden container to make an `axe` tests for `MultiActionButton` and not the `Accordion`
2. We should add one test to check expanded `MultiActionButton` with all buttons visible / rendered

**Pull request merged**